### PR TITLE
feat: add Discourse discussion comments

### PR DESCRIPTION
**Pull request by bonustrack**
Add Discourse discussion comments

Closes https://github.com/snapshot-labs/workflow/issues/111

image

**Testing**
- Check a proposal with a Discourse discussion like this: http://localhost:8080/#/s:safe.eth/proposal/0x86e0cee860e17a2f90b0d26e6e5ae0aff0c21330b8729ad449650c7ab84a0636/discussion
- Check a proposal without discussion:
https://snapshot.box/#/s:shot.eth/proposal/0x2835818fa8de4387f393dd52cca279f796b0b4ea4a3627021702c0e1e4b8782f
- Check proposal with a discussion that is not Discourse based:
http://localhost:8080/#/s:mcv.eth/proposal/0x41a560e5b09ffb39128a5170da00cca334ec162c2ae5039312a1759e0934d922

**TO-DO**

- [ ] Move to sidekick for requests to Discourse

### CODE REVIEW

**Comment by ChaituVR** [comment]
I get this error from https://corsproxy.io/

```JSON
{
  "error":{
    "code":403,
    "message":"Country blocked! Your country is blocked from accessing this ressource!"
  }
}
```

**Comment by bonustrack** [comment]
> I get this error from https://corsproxy.io/
> 
> ```json
> {
>   "error":{
>     "code":403,
>     "message":"Country blocked! Your country is blocked from accessing this ressource!"
>   }
> }
> ```

Right, forgot about that, I'll remove this to use sidekick service instead, meanwhile you can try with a VPN.

**Comment by wa0x6e** [comment]
This page https://bafybeibuwfiaag7wlyksn2d6kbe4n3bqjxvvge3j7dtkayvbfnhiy4bad4.on.fleek.co/#/s:mcv.eth/proposal/0x41a560e5b09ffb39128a5170da00cca334ec162c2ae5039312a1759e0934d922

is now throwing an error in the console

```
discourse.ts:21 
        
        
       GET https://corsproxy.io/?https%3A%2F%2Fhideout.metacartel.xyz%2Ft%2F307.json 403 (Forbidden)
```

**Comment by wa0x6e** [comment]
> > I get this error from https://corsproxy.io/
> > ```json
> > {
> >   "error":{
> >     "code":403,
> >     "message":"Country blocked! Your country is blocked from accessing this ressource!"
> >   }
> > }
> > ```
> 
> Right, forgot about that, I'll remove this to use sidekick service instead, meanwhile you can try with a VPN.

I got another type of error:

```
{
    "errors": [
        "You need to be logged in to do that."
    ],
    "error_type": "not_logged_in",
    "extras": {
        "title": "Page Not Found",
        "html": "\n\n\n\n<div class=\"page-not-found\">\n  <h1 class=\"title\">Oops! That page is private.</h1>\n\n    <a href=\"/login\" class='btn btn-primary'><svg class=\"fa d-icon svg-icon svg-node\" aria-hidden=\"true\"><svg id=\"user\" viewBox=\"0 0 448 512\">\n  <path d=\"M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z\"/>\n</svg></svg>\nLog In</a>\n\n</div>\n\n\n\n\n\n",
        "group": null
    }
}
```

**Comment by wa0x6e** [comment]
On this page https://bafybeibuwfiaag7wlyksn2d6kbe4n3bqjxvvge3j7dtkayvbfnhiy4bad4.on.fleek.co/#/s:safe.eth/proposal/0x1b48a83c44e323275a605b244a05bde89918fb9ec86be7bb83792eb26e544441/discussion

I am just getting 

```
Access to fetch at 'https://forum.safe.global/t/1131/posts.json' (redirected from 'https://corsproxy.io/?https%3A%2F%2Fforum.gnosis-safe.io%2Ft%2F1131%2Fposts.json') from origin 'https://bafybeibuwfiaag7wlyksn2d6kbe4n3bqjxvvge3j7dtkayvbfnhiy4bad4.on.fleek.co' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
```

**Comment by bonustrack** [comment]
> > > I get this error from https://corsproxy.io/
> > > ```json
> > > {
> > >   "error":{
> > >     "code":403,
> > >     "message":"Country blocked! Your country is blocked from accessing this ressource!"
> > >   }
> > > }
> > > ```
> > 
> > 
> > Right, forgot about that, I'll remove this to use sidekick service instead, meanwhile you can try with a VPN.
> 
> I got another type of error:
> 
> ```
> {
>     "errors": [
>         "You need to be logged in to do that."
>     ],
>     "error_type": "not_logged_in",
>     "extras": {
>         "title": "Page Not Found",
>         "html": "\n\n\n\n<div class=\"page-not-found\">\n  <h1 class=\"title\">Oops! That page is private.</h1>\n\n    <a href=\"/login\" class='btn btn-primary'><svg class=\"fa d-icon svg-icon svg-node\" aria-hidden=\"true\"><svg id=\"user\" viewBox=\"0 0 448 512\">\n  <path d=\"M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z\"/>\n</svg></svg>\nLog In</a>\n\n</div>\n\n\n\n\n\n",
>         "group": null
>     }
> }
> ```
> 
> From http://localhost:8080/#/s:mcv.eth/proposal/0x41a560e5b09ffb39128a5170da00cca334ec162c2ae5039312a1759e0934d922

This is just because the forum is private

**Comment by bonustrack** [comment]
Should be working fine now

File: apps/ui/src/helpers/utils.ts [hunk 1]
```diff
@@ -547,3 +547,8 @@ export function getFormattedVotingPower(votingPower?: VotingPowerItem) {
 
   return symbol ? `${value} ${symbol}` : value;
 }
+
+export function stripHtmlTags(text: string) {
+  const doc = new DOMParser().parseFromString(text, 'text/html');
+  return doc.body.textContent || '';
+}
```
[the remark this answers is not in the document]
Comment (bonustrack) [reply apps/ui/src/helpers/utils.ts]: Done

File: apps/ui/src/views/Proposal/Discussion.vue [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+</script>
+
+<template>
+  <div>
+    <div v-if="discussion" class="p-4 bg-skin-bg border-b">
+      <div class="max-w-[680px] mx-auto">
+        <a :href="discussion" target="_blank">
+          <UiButton class="flex items-center gap-2 w-full justify-center">
+            <component :is="ICDiscourse" class="size-[22px]" />
+            Join discussion
```
Comment (bonustrack) [inline apps/ui/src/views/Proposal/Discussion.vue:34]:
```suggestion
            Join the discussion
```

**Review by Sekhmet — APPROVED** [review]
tACK

**Pull request merged**